packages/react-components/react-provider/src/components/FluentProvider/useFluentProvider.ts

```diff
@@ -78,17 +82,31 @@ export const useFluentProvider_unstable = (
     theme: mergedTheme,
     // eslint-disable-next-line @typescript-eslint/naming-convention
     overrides_unstable: mergedOverrides,
-    themeClassName: useFluentProviderThemeStyleTag({ theme: mergedTheme, targetDocument }),
+    themeClassName: styleTagId,
 
     components: {
       root: 'div',
+      serverStyle: 'style',
     },
 
     root: getNativeElementProps('div', {
       ...props,
       dir,
       ref: useMergedRefs(ref, useFocusVisible<HTMLDivElement>({ targetDocument })),
     }),
+
+    serverStyle: resolveShorthand<React.HTMLAttributes<HTMLElement> & { 'data-fui-theme'?: '' }>(
+      undefined as undefined | Slot<'style'>,
+      {
+        required: !canUseDOM(),
+        defaultProps: {
+          [FUI_THEME_STYLE_ATTR]: '',
+          id: styleTagId,
+          dangerouslySetInnerHTML: { __html: rule },
+          ...renderer.styleElementAttributes,
+        },
+      },
+    ),
```

Comment:
```suggestion
    serverStyle: canUseDOM() ? undefined : {
          [FUI_THEME_STYLE_ATTR]: '',
          id: styleTagId,
          dangerouslySetInnerHTML: { __html: rule },
          ...renderer.styleElementAttributes,
      },
```

Let's avoid useless work

Reply:
I removed the entire slot in https://github.com/microsoft/fluentui/pull/27277/commits/f795b0858f347f2f9df0b88cd76f8567c67474b7

and went for inline rendering